omegaconf/dictconfig.py

```diff
@@ -331,7 +338,7 @@ def __getattr__(self, key: str) -> Any:
         except Exception as e:
             self._format_and_raise(key=key, value=None, cause=e)
 
-    def __getitem__(self, key: Union[str, Enum]) -> Any:
+    def __getitem__(self, key: Union[str, int, Enum]) -> Any:
```

Comment:
We should probably create an alias for the union instead of repeating the whole thing many times.